Common/include/geometry/dual_grid/CPoint.hpp

```diff
@@ -28,756 +28,774 @@
-            *Coord_Old,               /*!< \brief Old coordinates vector for primal solution reloading for Disc.Adj. with dynamic grid. */
-            *Coord_Sum,               /*!< \brief Sum of coordinates vector for geometry smoothing. */
-            *Coord_n,                 /*!< \brief Coordinates at time n for use with dynamic meshes. */
-            *Coord_n1,                /*!< \brief Coordinates at time n-1 for use with dynamic meshes. */
-            *Coord_p1;                /*!< \brief Coordinates at time n+1 for use with dynamic meshes. */
-  su2double *GridVel;                 /*!< \brief Velocity of the grid for dynamic mesh cases. */
-  su2double **GridVel_Grad;           /*!< \brief Gradient of the grid velocity for dynamic meshes. */
-  unsigned long Parent_CV;            /*!< \brief Index of the parent control volume in the agglomeration process. */
-  unsigned short nChildren_CV;        /*!< \brief Number of children in the agglomeration process. */
-  vector<unsigned long> Children_CV;  /*!< \brief Index of the children control volumes in the agglomeration process. */
-  bool Agglomerate_Indirect,          /*!< \brief This flag indicates if the indirect points can be agglomerated. */
-  Agglomerate;                        /*!< \brief This flag indicates if the element has been agglomerated. */
-  bool Move;                          /*!< \brief This flag indicates if the point is going to be move in the grid deformation process. */
-  unsigned long color;                /*!< \brief Color of the point in the partitioning strategy. */
-  su2double Wall_Distance;            /*!< \brief Distance to the nearest wall. */
-  su2double SharpEdge_Distance;       /*!< \brief Distance to a sharp edge. */
-  su2double Curvature;                /*!< \brief Value of the surface curvature (SU2_GEO). */
-  unsigned long GlobalIndex;          /*!< \brief Global index in the parallel simulation. */
-  unsigned short nNeighbor;           /*!< \brief Number of neighbors. */
-  bool Flip_Orientation;              /*!< \brief Flip the orientation of the normal. */
-  su2double MaxLength;                /*!< \brief The maximum cell-center to cell-center length. */
-  int *AD_InputIndex,                 /*!< \brief Indices of Coord variables in the adjoint vector. */
-  *AD_OutputIndex;                    /*!< \brief Indices of Coord variables in the adjoint vector after having been updated. */
+  const unsigned long nDim = 0;
+
+  su2vector<unsigned long> GlobalIndex;   /*!< \brief Global index in the parallel simulation. */
+
+  CCompressedSparsePatternUL Point;       /*!< \brief Points surrounding the central node of the control volume. */
+  CCompressedSparsePatternL Edge;         /*!< \brief Edges that set up a control volume (same sparse structure as Point). */
+  CCompressedSparsePatternL Elem;         /*!< \brief Elements that set up a control volume around a node. */
```

Comment:
Significant change no 1a: Adjacency information is stored in compressed format (instead of vector of vectors).